backend/inspirehep/pidstore/minters/base.py

```diff
@@ -78,11 +79,42 @@ def mint(cls, object_uuid, data):
 
     @classmethod
     def update(cls, object_uuid, data):
-        cls.mint(object_uuid, data)
+        minter = cls(object_uuid, data)
+        minter.validate()
+        pids_in_db = {
+            pid[0]
+            for pid in PersistentIdentifier.query.filter_by(
+                object_uuid=object_uuid,
+                pid_type=minter.pid_type,
+                object_type=minter.object_type,
+            )
+            .filter(PersistentIdentifier.status != PIDStatus.DELETED)
+            .with_entities("pid_value")
+            .all()
+        }
+        pids_requested = minter.get_pid_values()
+        pids_to_delete = pids_in_db - pids_requested
+        pids_to_create = pids_requested - pids_in_db
+        minter.delete(object_uuid, None, pids_to_delete)
```

Comment:
If we don't need to pass `data` anymore let's change the signature of delete, it's used only internaly.

Reply:
we still use data in `delete` methods in `ControlNumberMinter` and `CNUMMinter`